### PR TITLE
Ref #47941 added translations for Norma mail.

### PR DESCRIPTION
**Pull request by ShamryloMykola**

### CODE REVIEW

File: src/Plugin/rest/resource/ContractCancelResource.php [hunk 1]
```diff
@@ -48,6 +48,23 @@ class ContractCancelResource extends ResourceBase {
...
+    'first name' => 'Vorname',
+    'last name' => 'Nachname',
+    'email adress' => 'E-Mail Adresse',
+    'customer ID' => 'Kundennummer',
+    'sim card number' => 'SIM-Kartennummer',
+    'mobile phone number' => 'Mobilnummer',
+    'date of termination' => 'Kündigungsdatum',
+    'ordinary termination' => 'Ordentliche Kündigung',
+    'extraordinary termination' => 'Außerordentliche Kündigung',
+    'iban' => 'Iban',
```
Comment (boddy-devbranch) [inline src/Plugin/rest/resource/ContractCancelResource.php]: Please use t() for translations

**Review by boddy-devbranch — CHANGES_REQUESTED** [review]

File: src/Plugin/rest/resource/ContractCancelResource.php (hunk 1 above)
Comment (ShamryloMykola) [reply src/Plugin/rest/resource/ContractCancelResource.php]: Done!

**Review by boddy-devbranch — APPROVED** [review]

File: src/Plugin/rest/resource/ContractCancelResource.php [hunk 2]
```diff
@@ -285,9 +287,29 @@ public function post($data) {
...
+        'last name' => $this->t('Nachname'),
+        'email adress' => $this->t('E-Mail Adresse'),
+        'customer ID' => $this->t('Kundennummer'),
+        'sim card number' => $this->t('SIM-Kartennummer'),
+        'mobile phone number' => $this->t('Mobilnummer'),
+        'date of termination' => $this->t('Kündigungsdatum'),
+        'ordinary termination' => $this->t('Ordentliche Kündigung'),
+        'extraordinary termination' => $this->t('Außerordentliche Kündigung'),
+        'iban' => $this->t('Iban'),
+      ];
```
Comment (andrew-sereda) [inline src/Plugin/rest/resource/ContractCancelResource.php:308]: What is the point of using t() on translated strings?
Only English language can be used inside t() and translations can be added through PO file.
https://www.drupal.org/community/contributor-guide/reference-information/localize-drupal-org/working-with-offline/po-and-pot-files#s-custom-module-translation

**Review by andrew-sereda — CHANGES_REQUESTED** [review]

File: src/Plugin/rest/resource/ContractCancelResource.php (hunk 2 above)
Comment (ShamryloMykola) [reply src/Plugin/rest/resource/ContractCancelResource.php:308]: Done!

File: src/Plugin/rest/resource/ContractCancelResource.php [hunk 3]
```diff
@@ -285,9 +287,29 @@ public function post($data) {
 
       $body = '';
 
-      if (in_array($data['client'], ['norma', 'kaufland'])) {
-        foreach ($data as $data_key => $data_value) {
-          $body .= sprintf('<p>%s: %s</p>' . PHP_EOL, $data_key, $data_value);
+      $address = implode(' ', [$data['street'], $data['street number'], $data['zipcode'], $data['city']]);
+
+      if ($address) {
+        $body .= sprintf('<p>%s: %s</p>' . PHP_EOL, $this->t('Adresse'), $address);
```
Comment (andrew-sereda) [inline src/Plugin/rest/resource/ContractCancelResource.php]: Remove t() from here as well

**Review by andrew-sereda — CHANGES_REQUESTED** [review]

**Review by andrew-sereda — APPROVED** [review]

**Pull request merged**